pydantic_ai_slim/pydantic_ai/format_prompt.py

```diff
@@ -63,11 +77,24 @@ def format_as_xml(
 
 @dataclass
 class _ToXml:
+    data: Any
     item_tag: str
     none_str: str
-
-    def to_xml(self, value: Any, tag: str | None) -> ElementTree.Element:
-        element = ElementTree.Element(self.item_tag if tag is None else tag)
+    include_field_info: bool
+    repeat_field_info: bool
+    # a map of Pydantic Field paths to their metadata: a field unique string representation and its class
+    _fields: dict[str, tuple[str, FieldInfo | ComputedFieldInfo]] | None = None
+    # keep track of fields we have extracted attributes from
+    _parsed_fields: set[str] = field(default_factory=set)
+    # keep track of class names for dataclasses and Pydantic models, that occur in lists
+    _element_names: dict[str, str] | None = None
+    _FIELD_ATTRIBUTES = ('title', 'description', 'alias')
+
+    def to_xml(self, tag: str | None) -> ElementTree.Element:
+        return self._to_xml(self.data, tag)
+
+    def _to_xml(self, value: Any, tag: str | None, path: str = '') -> ElementTree.Element:
```

Comment:
If `path` should only be omitted for the root node, I think we should make it required and pass `''` explicitly there